dashboard/app/models/levels/javalab.rb

```diff
@@ -52,6 +52,12 @@ def fix_examples
     self.examples = examples.select(&:present?)
   end
 
+  def assign_attributes(params)
+    params[:visible].keys.map(&:to_s).each do |filename|
+      start_sources[filename]["visible"] = params[:visible][filename] === "true"
```

Comment:
What should happen if `start_sources[filename]` or `params[:visible][filename]` doesn't exist?